docs/DeveloperGuide.md

```diff
@@ -237,11 +237,25 @@ The following activity diagram summarizes what happens when a user executes a ne
   - Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
   - Cons: We must ensure that the implementation of each individual command are correct.
 
-_{more aspects and alternatives to be added}_
+### Delete feature
 
-### \[Proposed\] Data archiving
+#### Implementation
+This section explains the implementation of the `delete` feature. The command takes in one parameter which is the employee ID, executing the command leads to the removal of the employee with that specific employee ID from coydir.
 
-_{Explain here how the data archiving feature will be implemented}_
+Below is a sequence diagram and explanation of how `delete` is executed.
+
+<img src="images/DeleteCommandUML.png" width="550" />
+
+Step 1. The user enters the command `delete 1`
+
+Step 2. User input is parsed by `DeleteCommandParser`, then the method LogicManager#execute is called to create the DeleteCommand object.
```

Comment:
Perhaps can go a bit more into detail. Like DeleteCommandParser creates the DeleteCommand object while the execute function in DeleteCommand creates a CommandResult Object